asllib/AST.mli

```diff
@@ -397,6 +400,9 @@ type decl_desc =
   | D_Func of func
   | D_GlobalStorage of global_decl
   | D_TypeDecl of identifier * ty * (identifier * field list) option
+  | D_Pragma of identifier * expr list
+      (** A global pragma, as an explicit node to be used by tools which need
+          AST level hints about adjacent code. *)
```

Comment:
We do have line number annotation, so we do retain some idea of proximity.

Pragmas' only function is purely to be a code hint as far as I understand them. Them being reordered for type checking in aslref should be okay so long as a tool can parse them and perform some action using them in the untyped ast context (directly out of a parse). Once we get to typing and evaluation - they lose their relevance. Type checking their expressions is just a sensible thing to do for a sanity check, I think.

If you want I can remove the `about adjacent code` part of the comment to avoid confusion?